packages/wallet/backend/tests/incomingPayment/service.test.ts

```diff
@@ -0,0 +1,190 @@
+      )
+  }
+
+  beforeAll(async (): Promise<void> => {
+    bindings = await createContainer(env)
+    appContainer = await createApp(bindings)
+    knex = appContainer.knex
+    authService = await bindings.resolve('authService')
+    incopmPaymentService = await bindings.resolve('incomingPaymentService')
+
+    await mockIncomePaymentDeps()
+  })
+
+  beforeEach(async (): Promise<void> => {
+    const extraUserArgs = {
+      isEmailVerified: true,
+      rapydWalletId: 'mocked'
+    }
+
+    const { user } = await loginUser({
+      authService,
+      extraUserArgs
+    })
+    userId = user.id
+  })
+
+  afterAll(async (): Promise<void> => {
+    await appContainer.stop()
+    await knex.destroy()
+  })
+
+  afterEach(async (): Promise<void> => {
+    await truncateTables(knex)
+  })
+
+  describe('Create', () => {
+    it('should create an IncomePayment successfully', async () => {
+      const createdId = faker.string.uuid()
+      await mockIncomePaymentDeps(createdId)
+      const { walletAddress } = await prepareIncomePaymentDependencies()
+      const result = await incopmPaymentService.create(
+        userId,
+        walletAddress.id,
+        100
+      )
+      expect(result).toEqual(createdId)
+    })
+
+    it('should return NotFound Err', async () => {
+      await expect(
+        incopmPaymentService.create(userId, faker.string.uuid(), 100)
+      ).rejects.toThrowError(NotFound)
+    })
+  })
+
+  describe('Get PaymentDetails By Url', () => {
+    it('should get payment details successfully', async () => {
+      const { walletAddress, account } =
+        await prepareIncomePaymentDependencies()
+      const paymentId = faker.string.uuid()
+      const transaction = await Transaction.query().insert(
+        generateMockedTransaction({
+          walletAddressId: walletAddress.id,
+          accountId: account.id,
+          paymentId
+        })
+      )
+      const url = `${faker.internet.url()}/${paymentId}`
+      const result = await incopmPaymentService.getPaymentDetailsByUrl(url)
+      expect(result).toMatchObject({
+        description: transaction.description,
+        assetCode: transaction.assetCode
+      })
+    })
+
+    it('should throw err with wrong payment id', async () => {
+      const { walletAddress, account } =
+        await prepareIncomePaymentDependencies()
+      await Transaction.query().insert(
+        generateMockedTransaction({
+          walletAddressId: walletAddress.id,
+          accountId: account.id
+        })
+      )
+      const url = `${faker.internet.url()}/${faker.string.uuid()}`
+      await expect(
+        incopmPaymentService.getPaymentDetailsByUrl(url)
+      ).rejects.toThrowError(
+        /The provided incoming payment URL could not be found./
+      )
+    })
+  })
+  describe('getExternalPayment', () => {
+    beforeAll(async (): Promise<void> => {
+      axios.get = jest.fn().mockImplementation(() =>
```

Comment:
I think it will be better to mock a specific url response and test the difference between dev and other environments